include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -33,6 +37,92 @@ namespace execution
 inline namespace __dpl
 {
 
+namespace __internal
+{
+
+#if _ONEDPL_PREDEFINED_POLICIES
+struct __global_instance_tag {};
+#endif
+using _QueueFactory = sycl::queue (*)();
+
+class alignas(sycl::queue) __queue_holder
+{
+    static_assert(sizeof(sycl::queue) >= sizeof(std::pair<uintptr_t, _QueueFactory>));
+    static_assert(alignof(sycl::queue) >= alignof(std::uintptr_t));
+
+    union
+    {
+        sycl::queue __q;
+        std::pair<uintptr_t, _QueueFactory> __flag_and_factory;
```

Comment:
I don't see that you use it as a pointer (like it was in previous versions of implementation). Should it still be `uintptr_t`? Maybe `bool` is good enough? (FWIW, I would not care about padding or swap `first` and `second`)

Reply:
No, `bool` is not good enough. We want to check all bytes of the pointer not being zero, and not just a single byte.

Reply:
I got it, thanks